airbyte-cdk/python/airbyte_cdk/sources/streams/concurrent/state_converters/datetime_stream_state_converter.py

```diff
@@ -81,26 +82,34 @@ def convert_from_sequential_state(self, cursor_field: CursorField, stream_state:
         {
             "state_type": ConcurrencyCompatibleStateType.date_range.value,
             "metadata": { … },
+            "start": <timestamp representing the latest date before which all records were synced>
             "slices": [
-                {starts: 0, end: "2021-01-18T21:18:20.000+00:00", finished_processing: true}]
+                {"start": 0, "end": "2021-01-18T21:18:20.000+00:00"},
+            ]
         }
         """
         if self.is_state_message_compatible(stream_state):
             return stream_state
+        low_water_mark = (
+            self.parse_timestamp(stream_state[cursor_field.cursor_field_key])
+            if cursor_field.cursor_field_key in stream_state
+            else self.zero_value
+        )
         if cursor_field.cursor_field_key in stream_state:
```

Comment:
We have the condition `if cursor_field.cursor_field_key in stream_state` both in the definition of `low_water_mark` and this line. Can it be grouped up?